test/parallel/test-runner-exit-code.js

```diff
@@ -2,8 +2,28 @@
 const common = require('../common');
 const fixtures = require('../common/fixtures');
 const assert = require('assert');
-const { spawnSync } = require('child_process');
-const { setTimeout } = require('timers/promises');
+const { spawnSync, spawn } = require('child_process');
+const { once } = require('events');
+const { finished } = require('stream/promises');
+
+async function runAndKill(file) {
+  let stdout = '';
+  const child = spawn(process.execPath, ['--test', file]);
+  child.stdout.setEncoding('utf8');
+  child.stdout.on('data', (chunk) => {
+    if (!stdout.length) child.kill('SIGINT');
+    stdout += chunk;
+  });
+  const { 0: { 0: code, 1: signal } } = await Promise.all([once(child, 'exit'), finished(child.stdout)]);
```

Comment:
Do we need to wait for `finished(child.stdout)`? I would expect that once the child has exited, its stdout would be closed anyway.
```suggestion
  const [code, signal] = await once(child, 'exit');
```

Reply:
Surprisingly, we do need to wait,
see https://github.com/nodejs/node/issues/43618
and my pr that fixed it